src/transformers/models/marian/modeling_marian.py

```diff
@@ -1092,12 +1103,32 @@ def __init__(self, config: MarianConfig):
         self.post_init()
 
     def get_input_embeddings(self):
-        return self.shared
+        # This will return shared embeddings if they are shared else specific to encoder.
+        return self.get_encoder().get_input_embeddings()
 
     def set_input_embeddings(self, value):
-        self.shared = value
-        self.encoder.embed_tokens = self.shared
-        self.decoder.embed_tokens = self.shared
+        if self.config.share_encoder_decoder_embeddings:
+            self.shared = value
+            self.encoder.embed_tokens = self.shared
+            self.decoder.embed_tokens = self.shared
+        else:  # if not shared only set encoder embeedings
+            self.encoder.embed_tokens = value
+
+    def get_decoder_input_embeddings(self):
+        if self.config.share_encoder_decoder_embeddings:
+            raise ValueError(
```

Comment:
Why raise an error here? It's totally fine to just return `self.get_input_embeddigs()` in this case no?

Reply:
Still don't think we need to raise here ;-)